packages/@react-spectrum/s2/src/Picker.tsx

```diff
@@ -299,14 +337,25 @@ function Picker<T extends object>(props: PickerProps<T>, ref: FocusableRef<HTMLB
               placement={`${direction} ${align}` as Placement}
               shouldFlip={shouldFlip}
               UNSAFE_style={{
-                width: menuWidth ? `${menuWidth}px` : undefined
+                width: menuWidth && !isQuiet ? `${menuWidth}px` : undefined
               }}
               styles={style({
+                marginStart: {
+                  isQuiet: -12
+                },
                 minWidth: {
-                  default: '[var(--trigger-width)]'
+                  default: '[var(--trigger-width)]',
+                  isQuiet: 192
                 },
-                width: '[var(--trigger-width)]'
-              })}>
+                width: {
+                  default: '[var(--trigger-width)]',
+                  isQuiet: '[calc(var(--trigger-width) + var(--quietOffset))]'
+                },
+                '--quietOffset': {
+                  type: 'width',
+                  value: 24
+                }
```

Comment:
can maybe simplify with?
```suggestion
                  isQuiet: '[calc(var(--trigger-width) + (-1 * self(marginStart))]'
                }
```